api/envoy/extensions/filters/http/jwt_authn/v3/config.proto

```diff
@@ -230,6 +230,40 @@ message JwtProvider {
   //
   string payload_in_metadata = 9;
 
+  // If non empty, similar to :ref:`payload <envoy_v3_api_field_extensions.filters.http.jwt_authn.v3.JwtProvider.payload_in_metadata>`,
+  // a successfully verified JWT header will be written to :ref:`Dynamic State <arch_overview_data_sharing_between_filters>`
+  // as an entry (``protobuf::Struct``) in **envoy.filters.http.jwt_authn** *namespace* with the
+  // value of this field as the key.
+  //
+  // For example, if ``header_in_metadata`` is *my_header*:
+  //
+  // .. code-block:: yaml
+  //
+  //   envoy.filters.http.jwt_authn:
+  //     my_header:
+  //       alg: JWT
+  //       kid: EF71iSaosbC5C4tC6Syq1Gm647M
+  //       alg: PS256
+  //
+  // When the metadata has **envoy.filters.http.jwt_authn** entry already (for example if
+  // :ref:`payload_in_metadata <envoy_v3_api_field_extensions.filters.http.jwt_authn.v3.JwtProvider.payload_in_metadata>`
+  // is not empty), it will be inserted as a new entry in the same *namespace* as shown below:
+  //
+  // .. code-block:: yaml
+  //
+  //   envoy.filters.http.jwt_authn:
+  //     my_payload:
+  //       iss: https://example.com
+  //       sub: test@example.com
+  //       aud: https://example.com
+  //       exp: 1501281058
+  //     my_header:
+  //       alg: JWT
+  //       kid: EF71iSaosbC5C4tC6Syq1Gm647M
+  //       alg: PS256
+  //
+  string header_in_metadata = 14;
```

Comment:
what if `payload_in_metadata` is the same as `header_in_metadata`? Is it allowed?

Reply:
it should not be allowed, please add comment on this restriction.

Reply:
Added a warning section. Please let me know if we want to validate this when checking the config.

Reply:
hmm, the current comment says `... is not allowed ...` which I would expect the validation is going to reject the config, or you can just update the comment to be more accurate (it's not suggested due to potential override but can still be used (won't be rejected in validation) if the user is sure it won't have any override for their use case).

Reply:
@yangminzhu I updated as you suggested.